src/components/CatalogPage/CatalogPage.tsx

```diff
@@ -0,0 +1,190 @@
+import { FormEvent, useContext, useEffect, useState } from 'react';
+import { FilterSection } from '../FilterSection/FilterSection';
+import { SortSection } from '../SortSection/SortSection';
+import { IsLoadindContext } from '../../App';
+import { ProductService } from '../../services';
+import { CTResponse } from '../../ct-client';
+import {
+  Category,
+  ProductProjectionPagedQueryResponse,
+} from '@commercetools/platform-sdk';
+import { ProductPreviewItem } from '../ProductPreviewItem/ProductPreviewItem';
+import { convertPrice } from '../../utils/convertPrice';
+
+/* import { convertPrice } from '../../utils/convertPrice'; */
+
+interface ProductProjectionResponse {
+  productProjectionSearch: ProductProjectionPagedQueryResponse;
+}
+
+export const CatalogPage = () => {
+  const [showFilter, setShowFilter] = useState(false);
+  const [handleLoading] = useContext(IsLoadindContext);
+  const [currentSort, setCurrentSort] = useState<string>();
+  const [currentSearch, setcurrentSearch] = useState<string>();
+  const [currentCategories, setCurrentCategories] = useState<string[]>([]);
+
+  const [products, setProducts] = useState<ProductProjectionResponse>({
+    productProjectionSearch: {
+      limit: 0,
+      count: 0,
+      offset: 0,
+      results: [],
+    },
+  });
+
+  const [currentRangeValue, setCurrentRangeValue] = useState<number[]>([
+    0, 3100,
+  ]);
+  const handleRangeSlider = (event: number | number[]) => {
+    if (typeof event !== 'number') {
+      setCurrentRangeValue(event);
+    }
+  };
+
+  const handleCategories = (event: React.ChangeEvent) => {
+    const target = event.target as HTMLInputElement;
+    if (target.checked) {
+      setCurrentCategories([...currentCategories, target.value]);
+    } else {
+      currentCategories.splice(currentCategories.indexOf(target.value), 1);
+      setCurrentCategories([...currentCategories]);
+    }
+  };
+
+  const handleCurrentSort = (event: React.ChangeEvent) => {
+    const target = event.target as HTMLInputElement;
+    setCurrentSort(target.value);
+  };
+
+  const handleCurrentSearch = (event: FormEvent) => {
+    const target = event.target as HTMLInputElement;
+    setcurrentSearch(target.value);
+  };
+
+  useEffect(() => {
```

Comment:
you can implement that in a sortest way, like this 


useEffect(() => {
    const fetchProducts = async () => {
      handleLoading(true);
      const productService = new ProductService();
      let data: Promise<CTResponse>;

      if (
        currentSort ||
        currentSearch ||
        currentCategories.length > 0 ||
        currentRangeValue[0] !== 0 ||
        currentRangeValue[1] !== 3100
      ) {
        data = productService.getProductsWithFilters(
          'EN-GB',
          currentRangeValue,
          currentSort,
          currentSearch,
          currentCategories
        );
      } else {
        data = productService.getProductsAll();
      }

      const response = await data;
      handleLoading(false);
      setProducts(response.data as ProductProjectionResponse)